sw/device/silicon_creator/lib/drivers/hmac.c

```diff
@@ -78,3 +78,9 @@ void hmac_sha256_final(hmac_digest_t *digest) {
                         (i * sizeof(uint32_t)));
   }
 }
+
+void hmac_sha256(const void *data, size_t len, hmac_digest_t *digest) {
+  hmac_sha256_init();
+  hmac_sha256_update(data, len);
```

Comment:
You're right. IIRC we discussed this before and while this is UB as defined in the standard, we thought this usage was OK since no riscv or x64 compiler behaves differently here and this is slightly clearer. That being said, this was a long time ago and I'm happy to revisit this decision. @luismarques WDYT? This probably doesn't need an immediate resolution but if we end up using `char` instead of `uint8_t` we should add a short note to the style guide.

Reply:
I don't have strong opinions about this. Yes, using `char` directly would avoid a potential correctness issue, but it's one that probably won't ever apply to our platform. It probably also better signals that we are relying on the special properties of `char` to avoid aliasing issues. But if `uint8_t` is slightly clearer in some contexts I can see that trumping the other issue.

Reply:
Thanks, Luis. Maybe I'll document this in the style guide.